src/compiler/checker.ts

```diff
@@ -3177,6 +3182,12 @@ namespace ts {
                 }
                 return parentType;
             }
+            // In strict null checking mode, a default value of a binding pattern adds undefined,
+            // which should be removed to get the type of the elements
+            const func = getContainingFunction(declaration);
+            if (strictNullChecks && func && !func.body && getFalsyFlags(parentType) & TypeFlags.Undefined) {

```

Comment:
Not sure I understand the `func && !func.body` check. Also, where do you check if the binding pattern has a default value?